docs/src/basics/autodiff.md

```diff
@@ -0,0 +1,51 @@
+# Automatic Differentiation Backends
+
+## Summary of Finite Differencing Backends
+
+  - [`AutoFiniteDiff`](@ref): Finite differencing, not optimal but always applicable.
+  - [`AutoSparseFiniteDiff`](@ref): Sparse version of [`AutoFiniteDiff`](@ref).
+
+## Summary of Forward Mode AD Backends
+
+  - [`AutoForwardDiff`](@ref): The best choice for dense problems.
+  - [`AutoSparseForwardDiff`](@ref): Sparse version of [`AutoForwardDiff`](@ref).
+  - [`AutoPolyesterForwardDiff`](@ref): Might be faster than [`AutoForwardDiff`](@ref) for
+    large problems. Requires `PolyesterForwardDiff.jl` to be installed and loaded.
+
+## Summary of Reverse Mode AD Backends
+
+  - [`AutoZygote`](@ref): The fastest choice for non-mutating array-based (BLAS) functions.
+  - [`AutoSparseZygote`](@ref): Sparse version of [`AutoZygote`](@ref).
+  - [`AutoEnzyme`](@ref): Uses `Enzyme.jl` Reverse Mode and should be considered
+    experimental.
+
+!!! note
+    
+    If `PolyesterForwardDiff.jl` is installed and loaded, then `SimpleNonlinearSolve.jl`
+    will automatically use `AutoPolyesterForwardDiff` as the default AD backend.
+
```

Comment:
```suggestion
!!! note
    
    If `PolyesterForwardDiff.jl` is installed and loaded, then `SimpleNonlinearSolve.jl`
    will automatically use `AutoPolyesterForwardDiff` as the default AD backend.

!!! note

    The `sparse` versions of the methods refers to automated sparsity detection. These methods
    automatically discover the sparse Jacobian form from the function `f`. Note that all methods
    specialize the differentiation on a sparse Jacobian if the sparse Jacobian is given as
    `prob.f.jac_prototype` in the `NonlinearFunction` definition, and the `autosparse` here simply
    refers to whether this `jac_prototype` should be generated automatically. Note too that
    this sparsity detection has a non-trivial overhead and thus as an optimization it's recommended
    that one runs the sparsity detection outside of the solver loop if many solves are required.
    For more details, see [SparseDiffTools.jl](https://github.com/JuliaDiff/SparseDiffTools.jl)
```

Reply:
Should we link https://docs.sciml.ai/NonlinearSolve/stable/basics/sparsity_detection/ instead of SparseDiffTools directly?

Reply:
If `AutoForwardDiff` is specified and `sparsity` is specified, we still use non-color AD. However, if the `autodiff = nothing`, we will choose `AutoSparseForwardDiff`. So the note "specialize the differentiation on a sparse Jacobian if the sparse Jacobian" seems a bit incorrect.

The reason to do this is in the last warning of https://docs.sciml.ai/NonlinearSolve/stable/basics/sparsity_detection/ (mostly to allow an external override of any default internal selection algorithm)

Reply:
not for a performance consideration, it is to allow an external override. If only jac_prototype is specified, then sparse is used. But if the user manually decides to use AutoForwardDiff overriding the default autodiff choice, then I feel we should use the autodiff provided.

Reply:
I guess so, though that's an odd combination 😅. Can you reword this and merge?